File: packages/transform/src/features/notations/NotationCreateValidateGeneralTransformer.ts
import { NotationValidateGeneralProgrammer } from "@typia/core";
import { NamingConvention } from "@typia/utils";

import { ITransformProps } from "../../ITransformProps";
import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace NotationCreateValidateGeneralTransformer {
  export const transform =
    (rename: (str: string) => string) => (props: ITransformProps) =>
      GenericTransformer.factory({
        ...props,
        method: `notations.createValidate${NamingConvention.capitalize(rename.name)}`,
        write: (x) =>
          NotationValidateGeneralProgrammer.write({
            ...x,
            rename,
          }),
      });
}

File: packages/transform/src/features/notations/NotationCreateValidateGeneralTransformer.go
package notations

import (
	corenotations "github.com/samchon/typia/packages/core/src/programmers/notations"
	internal "github.com/samchon/typia/packages/transform/src/internal"
)

var NotationCreateValidateGeneralTransformer = notationCreateValidateGeneralTransformerNamespace{}

type notationCreateValidateGeneralTransformerNamespace struct{}

func (notationCreateValidateGeneralTransformerNamespace) Transform(rename string) internal.Task {
	return func(call internal.ITypiaCallExpression) string {
		return corenotations.NotationValidateGeneralProgrammer.Write(rename, call.TypeText)
	}
}
